common/remote/sockfile.cpp

```diff
@@ -2683,14 +2683,11 @@ class CRemoteKeyManager : public CSimpleInterfaceOf<IKeyManager>
             {
                 if (!first && (nullptr == keyCursor)) // No keyCursor implies there is nothing more to fetch
                     return false;
-                size32_t maxRecs = maxRecsPerRequest;
+                unsigned maxRecs = maxRecsPerRequest;
                 if (maxRecs && chooseNLimit)
                 {
-                    if (totalGot >= chooseNLimit)
-                        break;
-                    maxRecs = chooseNLimit - totalGot;
-                    if (maxRecs > maxRecsPerRequest)
-                        maxRecs = maxRecsPerRequest; // limit to
+                    if (totalGot + maxRecs > chooseNLimit)
+                        maxRecs = chooseNLimit - totalGot;
```

Comment:
picky: Should have a cast to unsigned otherwise the windows compiler will complain.  i.e.

(unsigned)(chooseNLimit - totalGo);